--- origsrc/mtpaint-3.21/configure	2008-06-07 20:58:46.000000000 -0500
+++ src/mtpaint-3.21/configure	2008-06-13 16:23:33.281250000 -0500
@@ -53,7 +53,7 @@
 		then
 			L_EX="${L_EX}.a"
 		else
-			L_EX="${L_EX}.so"
+			L_EX="${L_EX}.dll.a"
 		fi
 	else
 		if [ "$2" ]
--- origsrc/mtpaint-3.21/src/font.c	2008-05-29 16:08:16.000000000 -0500
+++ src/mtpaint-3.21/src/font.c	2008-06-13 16:23:33.359375000 -0500
@@ -613,7 +613,7 @@
 
 		if ( stat(full_name, &buf)<0 ) continue;	// Get file details
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 		if ( S_ISDIR(buf.st_mode) )
 #else
 		if ( ep->d_type == DT_DIR || S_ISDIR(buf.st_mode) )
